File: Haskell/p001-Basic-Neuron-3-inputs.hs
multWeights :: [Double] -> [Double] -> Double  -> Double
multWeights inputs weights bias = sum  (zipWith (*) inputs weights) + bias

inputs :: [Double]
inputs = [1.2, 5.1, 2.1]

weights :: [Double]
weights = [3.1, 2.1, 8.7]

bias :: Double
bias = 3.0

output :: Double
output = multWeights inputs weights bias

main = print output

File: Fortran/p001-Basic-Neuron-3-inputs.f90
program NN
    implicit none
    real :: inputs(3),weights(3),bias,output

    inputs  = [1.2, 5.1, 2.1]
    weights = [3.1, 2.1, 8.7]
    bias    = 3

    output = inputs(1) * weights(1) +&
             inputs(2) * weights(2) +&
             inputs(3) * weights(3) +&
             bias

    print *,output
end program NN
